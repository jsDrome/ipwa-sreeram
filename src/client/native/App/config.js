const domain = "https://ipwa-sreeram.web.app";

module.exports = {
  showBannerAd: false,
  showInterstitialAd: false,
  ad: {
    test: {
      bannerAdUnitId: 'ca-app-pub-3940256099942544/6300978111',
      interstitialAdUnitID: 'ca-app-pub-3940256099942544/4411468910',
    },
    production: {
      bannerAdUnitId: 'ca-app-pub-6831276331714408/9053954679',
      interstitialAdUnitID: 'ca-app-pub-6831276331714408/1135882305',
    },
  },
  webview: {
    userAgent: "Mozilla/5.0 (Linux; Android 7.0; SM-G930V Build/NRD90M) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36",
    uri: `${domain}?app=true`,
  },
  splashUri: `${domain}/img/splash.png`,
};
