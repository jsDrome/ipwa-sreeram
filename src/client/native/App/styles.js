export default {
  app: {
    flex: 1,
    backgroundColor: '#000000',
  },
  webView: {
    height: '100%',
    marginTop: 30,
  },
  loader: {
    backgroundColor: '#000000',
    marginTop: -30,
    width: '100%',
    height: '100%',
  },
  bannerAd: {
    position: "absolute",
    bottom: 0,
  },
};
