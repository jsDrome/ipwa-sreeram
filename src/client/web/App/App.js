import React, { Fragment, useEffect } from "react";
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Alert } from '@jsdrome/components';

import Meta from '../_components/Meta/Meta';
import { checkForMessage } from '../../../store/app/appActions';

const App = ({ children, checkForMessage, alert }) => {

  useEffect(checkForMessage, []);

  return <Fragment>
    <Meta />
    {alert && <Alert type={alert.type} duration={alert.duration}>{alert.message}</Alert>}
    {children}
  </Fragment>;
}

App.propTypes = {
  children: PropTypes.node,
};

App.defaultProps = {
  children: null,
};

const mapStateToProps = state => ({
  alert: alertSelector(state),
});

const alertSelector = state => state.app.alert;

const mapDispatchToProps = dispatch => ({
  checkForMessage: data => checkForMessage(data)(dispatch),
})

export {
  App,
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
