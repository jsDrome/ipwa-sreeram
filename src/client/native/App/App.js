import React, { useState, useEffect } from 'react';
import { View, StatusBar, ImageBackground } from 'react-native';
import { WebView } from 'react-native-webview';
import { AdMobBanner } from 'expo-ads-admob';
import { Notifications } from 'expo';
import * as Permissions from 'expo-permissions';
import * as Updates from 'expo-updates';

import styles from './styles';

const { showBannerAd, ad, webview: { uri, userAgent }, splashUri } = require('./config');

// eslint-disable-next-line no-undef
const env = __DEV__ ? "test" : "production"
const { bannerAdUnitId } = ad[env];

const App = () => {

  const [ backgroundColor, setBackgroundColor ] = useState('#000000');
  const [ pushToken, setPushToken ] = useState(null);

  useEffect(() => {
    setTimeout(() => {
      setBackgroundColor('#ffffff');
      // eslint-disable-next-line no-magic-numbers
    }, 4000);

    setToken();
    checkForUpdates();
  });

  const setToken = async () => {
    try {
      const token = await registerForPushNotificationsAsync();
      setPushToken(token);
    } catch (e) {
      // handle or log error
    }
  }

  const checkForUpdates = async () => {
    try {
      const update = await Updates.checkForUpdateAsync();
      if (update.isAvailable) {
        await Updates.fetchUpdateAsync();
        Updates.reloadFromCache();
      }
    } catch (e) {
      // handle or log error
    }
  }

  return <View style={styles.app}>
    <StatusBar barStyle="light-content" />
    <WebView
      userAgent={userAgent}
      javaScriptEnabled={true}
      domStorageEnabled={true}
      startInLoadingState={true}
      scrollEnabled={true}
      renderLoading={Loader}
      injectedJavaScript={`window.pushToken='${pushToken}'`}
      bounce={false}
      useWebKit={true}
      source={{ uri }}
      style={{ ...styles.webView, backgroundColor }} />
    {showBannerAd && <BannerAd />}
  </View>
};

const BannerAd = () => AdMobBanner && <AdMobBanner
  style={styles.bannerAd}
  bannerSize="fullBanner"
  adUnitID={bannerAdUnitId}
  testDeviceID="EMULATOR"
  servePersonalizedAds
  onDidFailToReceiveAdWithError={() => console.log('error from ad')} />;

const Loader = () => <ImageBackground source={{ uri: splashUri }} style={styles.loader} />;

const registerForPushNotificationsAsync = async () => {
  const { status: existingStatus } = await Permissions.getAsync(Permissions.NOTIFICATIONS);
  let finalStatus = existingStatus;

  if (existingStatus !== 'granted') {
    const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS);
    finalStatus = status;
  }

  if (finalStatus !== 'granted') return null;

  let token = null;

  try {
    token = await Notifications.getExpoPushTokenAsync();
  } catch (e) {
    console.log(e);
  }

  return token;
};

export default App;
