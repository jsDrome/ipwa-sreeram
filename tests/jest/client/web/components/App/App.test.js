import React from 'react';
import { App } from '@src/client/web/App/App';
import ShallowRenderer from 'react-test-renderer/shallow'

const renderer = new ShallowRenderer();

test('App -> render', () => {
  const component = renderer.render(<App />);
  expect(component).toMatchSnapshot();
});
