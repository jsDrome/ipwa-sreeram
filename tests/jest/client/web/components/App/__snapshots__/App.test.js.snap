// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App -> render 1`] = `
<React.Fragment>
  <MetaTags />
</React.Fragment>
`;
